src/foam/nanos/auth/AuthenticatedUserDAO.java

```diff
@@ -64,17 +63,36 @@ public FObject put_(X x, FObject obj) {
     if ( ! SafetyUtil.equals(toPut.getId(), user.getId()) &&
       ! auth.check(x, GLOBAL_USER_UPDATE) &&
       ! auth.check(x, GLOBAL_SPID_UPDATE) &&
-      ! auth.check(x, "spid.update." + user.getSpid()) ) {
-      throw new RuntimeException("Unable to update user");
+      ! auth.check(x, "spid.update." + user.getSpid())
+    ) {
+      throw new AuthorizationException("You do not have permission to update this user.");
     }
 
     // set spid if not set
-    if ( SafetyUtil.isEmpty(toPut.getSpid()) &&
-        ! SafetyUtil.isEmpty(user.getSpid()) ) {
-      toPut.setSpid(user.getSpid());
+    if ( SafetyUtil.isEmpty((String) newUser.getSpid()) &&
```

Comment:
Where are you getting the `newUser` from?

Reply:
I originally changed `toPut` to `newUser` and `existingUser` to `oldUser`. Some of those changes must have gotten lost, probably in merge conflict fixes.

Reply:
yeah I think thats whats happened